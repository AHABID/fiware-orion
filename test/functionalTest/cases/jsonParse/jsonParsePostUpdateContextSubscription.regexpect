1: ***********************************
HTTP/1.1 200 OK
Content-Length: 180
Content-Type: application/json
Date: REGEX(.*)

{
    "subscribeError": {
        "errorCode": {
            "code": "404",
            "reasonPhrase": "No context element found"
        },
        "subscriptionId": "REGEX([0-9a-f]{24})"
    }
}
2: ***********************************
HTTP/1.1 200 OK
Content-Length: 134
Content-Type: application/json
Date: REGEX(.*)

{
    "subscribeResponse": {
        "duration": "P5Y", 
        "subscriptionId": "REGEX([0-9a-f]{24})", 
        "throttling": "P5Y"
    }
}
3: ***********************************
HTTP/1.1 200 OK
Content-Length: 134
Content-Type: application/json
Date: REGEX(.*)

{
    "subscribeResponse": {
        "duration": "P4Y",
        "subscriptionId": "REGEX([0-9a-f]{24})",
        "throttling": "P4Y"
    }
}
4: ***********************************
