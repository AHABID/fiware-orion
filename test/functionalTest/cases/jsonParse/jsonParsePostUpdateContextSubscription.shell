echo "1: ***********************************"
url="/NGSI10/updateContextSubscription"
payload='{
  "duration": "P5Y",
  "restriction": {
    "attributeExpression" : "AE",
    "scopes": [
      {
        "type" : "t1",
        "value" : "v1"
      },
      {
        "type" : "t2",
        "value" : "v2"
      }
    ]
  },
  "subscriptionId": "9212ce4b0c214479be429e2b",
  "notifyConditions": [
    {
      "type": "ONCHANGE",
      "condValues": [
        "CondValue1",
        "CondValue2"
      ],
      "restriction": "restriction"
    }
  ],
  "throttling": "P5Y"
}'
curlJson ${url} "${payload}"

echo "2: ***********************************"
url="/NGSI10/subscribeContext"
payload='{
  "entities": [
    {
      "type": "Room",
      "isPattern": "false",
      "id": "ConferenceRoom"
    },
    {
      "type": "Room",
      "isPattern": "false",
      "id": "OfficeRoom"
    }
  ],
  "attributes": [
    "temperature",
    "occupancy",
    "lightstatus"
  ],
  "reference": "127.0.0.1",
  "duration": "P5Y",
  "restriction": {
    "attributeExpression": "testRestriction",
    "scopes" : [
      {
        "type" : "t1",
        "value" : "v1"
      },
      {
        "type" : "t2",
        "value" : "v2"
      }
    ]
  },
  "notifyConditions": [
    {
      "type": "ONCHANGE",
      "condValues": [
        "CondValue1",
        "CondValue2"
      ]
      ,
      "restriction": "restriction"
    }
  ],
  "throttling": "P5Y"
}'
curlJson ${url} "${payload}"

SUB_ID=$(echo "$response" | grep subscriptionId | awk -F\" '{ print $4}')

echo "3: ***********************************"
url="/NGSI10/updateContextSubscription"
payload='{
  "duration": "P4Y",
  "restriction": {
    "attributeExpression" : "AE",
    "scopes": [
      {
        "type" : "t1",
        "value" : "v1"
      },
      {
        "type" : "t2",
        "value" : "v2"
      }
    ]
  },
  "subscriptionId": "'${SUB_ID}'",
  "notifyConditions": [
    {
      "type": "ONCHANGE",
      "condValues": [
        "CondValue1",
        "CondValue2"
      ],
      "restriction": "restriction"
    }
  ],
  "throttling": "P4Y"
}'
curlJson ${url} "${payload}"

echo "4: ***********************************"
