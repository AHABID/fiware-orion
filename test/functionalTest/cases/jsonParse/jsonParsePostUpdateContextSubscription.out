1: ***********************************
HTTP/1.1 200 OK
Content-Length: 180
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:21:57 GMT

{
    "subscribeError": {
        "errorCode": {
            "code": "404", 
            "reasonPhrase": "No context element found"
        }, 
        "subscriptionId": "9212ce4b0c214479be429e2b"
    }
}
2: ***********************************
HTTP/1.1 200 OK
Content-Length: 134
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:21:58 GMT

{
    "subscribeResponse": {
        "duration": "P5Y", 
        "subscriptionId": "538c342678bc2c993a59324c", 
        "throttling": "P5Y"
    }
}
3: ***********************************
HTTP/1.1 200 OK
Content-Length: 134
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:21:58 GMT

{
    "subscribeResponse": {
        "duration": "P4Y", 
        "subscriptionId": "538c342678bc2c993a59324c", 
        "throttling": "P4Y"
    }
}
4: ***********************************
